packages/functions/index.js

```diff
@@ -0,0 +1,14 @@
+/* global globalThis */
+
+export const waitUntil = promise => {
+  if (
+    promise === null ||
+    typeof promise !== 'object' ||
+    typeof promise.then !== 'function'
+  ) {
+    throw new TypeError('waitUntil can only be called with a Promise');
+  }
+
+  const ctx = globalThis[Symbol.for('@vercel/request-context')]?.get?.() ?? {};
+  ctx.waitUntil?.(promise);
```

Comment:
Avoid failing silently. Make it explicit.

```suggestion
  const ctx = globalThis[Symbol.for('@vercel/request-context')]?.get?.() ?? {};
  let waitUntil = ctx.waitUntil?;
  if (!waitUntil) {
    throw new Error('failed to get waitUntil function for this request context');
  }
  
  waitUntil(promise);
```